src/UI/Shared/WB.UI.Shared.Enumerator/OfflineSync/Services/Implementation/SecureStorage.cs

```diff
@@ -59,8 +42,14 @@ private void Save()
         /// <param name="dataBytes">Data bytes to store.</param>
         public void Store(string key, byte[] dataBytes)
         {
-            this.keyStore.SetEntry(key, new KeyStore.SecretKeyEntry(new SecureData(dataBytes)), null);
-            Save();
+            lock (keyStoreFileLock)
+            {
+                this.keyStore.SetEntry(key, new KeyStore.SecretKeyEntry(new SecureData(dataBytes)), null);
+                using var keyStoreFile = Application.Context.OpenFileOutput(keyStoreFileName, FileCreationMode.Private);
+                this.keyStore.Store(keyStoreFile, null);
+                
+                this.inMemoryKeyStorage.Remove(key);

```

Comment:
why do you remove cached entity? you can store new value of it.

Reply:
Discussed. Agreed to leave it.